main.py

```diff
@@ -53,3 +53,30 @@ def update_exercise(exercise_id: int, exercise: schemas.ExercisePatch, db: Sessi
         raise HTTPException(status_code=404, detail="exercise not found")
     return updated_exercise
 
+@app.post("/accounts/", response_model=schemas.AccountOut)
+def create_account(account_in: schemas.AccountIn,db: Session = Depends(get_db)):
+    account_saved= crud.save_user(db,account_in)
+    return account_saved
+
+@app.get("/accounts/", response_model=list[schemas.AccountOut])
+def get_all_accounts(db: Session = Depends(get_db)):
+    accounts = db.query(models.Account).all()
+    if not accounts:
+        return []
+    return accounts
+
+@app.delete("/accounts/{account_id}")
+def delete_account(account_id: int, db: Session = Depends(get_db)):
+    db_account = crud.get_account(db, account_id=account_id)
+    if db_account is None:
+        raise HTTPException(status_code=404, detail="account not found")
+    crud.delete_account(db=db, account_id=account_id)
+    return {"message": "account deleted"}
+
+
+@app.patch("/accounts/{account_id}")
+def update_account(account_id: int, account: schemas.AccountIn, db: Session = Depends(get_db)):
```

Comment:
@vvihorev requested on my code that we do the none-check here in the main.py and not in the crud.py. so for consistency i would recommend to do the same here

```
    account = crud.get_account(db, account_id=account_id)
    if account is None:
        raise HTTPException(status_code=404, detail="account not found")
    changed_account= crud.update_account(db, account_id=account_id, account =account )
    return changed_account
```